lib/taskana-core-test/src/test/java/acceptance/classification/delete/DeleteClassificationAccTest.java

```diff
@@ -0,0 +1,240 @@
+            .objectReference(DefaultTestEntities.defaultTestObjectReference().build())
+            .build();
+    TaskBuilder.newTask()
+        .classificationSummary(classification.asSummary())
+        .workbasketSummary(defaultWorkbasket)
+        .attachments(attachment)
+        .primaryObjRef(DefaultTestEntities.defaultTestObjectReference().build())
+        .buildAndStore(taskService);
+
+    assertThatThrownBy(
+            () -> classificationService.deleteClassification(attachmentClassification.getId()))
+        .isInstanceOf(ClassificationInUseException.class);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_ThrowExceptionAndRollback_When_ChildClassificationIsInUse() throws Exception {
+    Classification parent =
+        DefaultTestEntities.defaultTestClassification().buildAndStore(classificationService);
+    Classification child =
+        DefaultTestEntities.defaultTestClassification()
+            .parentId(parent.getId())
+            .buildAndStore(classificationService);
+    TaskBuilder.newTask()
+        .primaryObjRef(DefaultTestEntities.defaultTestObjectReference().build())
+        .classificationSummary(child)
+        .workbasketSummary(defaultWorkbasket)
+        .buildAndStore(taskService);
+
+    assertThatThrownBy(() -> classificationService.deleteClassification(parent.getId()))
+        .isInstanceOf(ClassificationInUseException.class);
+    Classification rollback =
+        classificationService.getClassification(parent.getKey(), parent.getDomain());
+    assertThat(rollback.getDomain()).isEqualTo(parent.getDomain());
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_ThrowExceptionAndRollbackAllDomains_When_ChildClassificationIsInUse()
+      throws Exception {
+    Classification parent =
+        DefaultTestEntities.defaultTestClassification().buildAndStore(classificationService);
+    Classification child =
+        DefaultTestEntities.defaultTestClassification()
+            .parentId(parent.getId())
+            .buildAndStore(classificationService);
+    TaskBuilder.newTask()
+        .primaryObjRef(DefaultTestEntities.defaultTestObjectReference().build())
+        .classificationSummary(child)
+        .workbasketSummary(defaultWorkbasket)
+        .buildAndStore(taskService);
+
+    assertThatThrownBy(() -> classificationService.deleteClassification(parent.getKey(), ""))
+        .isInstanceOf(ClassificationInUseException.class);
+    Classification rollbackMaster = classificationService.getClassification(parent.getKey(), "");
+    Classification rollbackDomain =
+        classificationService.getClassification(parent.getKey(), parent.getDomain());
+    assertThat(rollbackDomain.getKey()).isEqualTo(rollbackMaster.getKey());
+    assertThat(rollbackDomain.getDomain()).isNotEqualTo(rollbackMaster.getDomain());
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_ThrowClassificationNotFoundException_IfClassificationDoesNotExist() {
```

Comment:
Please rename ^^ Happens really frequent to me too!

Reply:
Ok